### DISCUSSION TITLE
Grasp Depth Question

### DISCUSSION THREAD

**Discussion by yuji01-dfki** [Victorlouisdg/cloth-competition#8] [Questions]
Dear Organizer Team,

I have a short question regarding the rule.

As of my understanding, the participants only submit the **grasp pose** for execution, since the .json file of grasp pose submission does not include the "grasp depth" term. Can I take the assumption that the **grasp depth** is set to a fixed value for everyone （e.g. the 5cm used in data collection) during developing? 

Thanks in advance and wish you a nice day,
Yuhan

**Comment by Victorlouisdg** [comment]
Hi Yuhan,

Thanks for reaching out with this important question! Grasp depth is not fixed, and participants are free to try different grasp depths. Here's how it works:

- **You control the grasp depth:** The grasp pose you provide will be used directly by the robot, it will be the TCP pose where the gripper is closed. This means you decide the appropriate grasp depth and include it in your pose calculation.

- **Dataset as a guide:** We used 5cm as the default grasp depth during data collection, which proved to be a good balance between the risk of missing grasps and the risk of grasping too much. This should also be a good starting point for your own grasps. In the dataset you can also find samples, such as `sample_000008` where the grasp annotator decided to add (or remove) a few cm to the grasp depth. 

- **Calculation code:** If you'd like to replicate our exact grasp pose calculations, check out the [`grasp_annotation.py`](https://github.com/Victorlouisdg/cloth-competition/blob/161b2c0744bc6075e94b578a098291860c026ed6/cloth-tools/cloth_tools/annotation/grasp_annotation.py) file. These two lines are key:
```python
Z = approach_direction / np.linalg.norm(approach_direction)
position_with_depth = position + grasp_depth * Z
```

I hope this clarifies things! Feel free to ask if you have any more questions.
